cmd/gcp-controller-manager/csr_approver.go

```diff
@@ -626,19 +626,35 @@ func clusterHasInstance(ctx *controllerContext, instanceZone string, instanceID
 	return false, nil
 }
 
+type foundError struct{}
+
+func (*foundError) Error() string{
+	return "found"
+}
+
 func groupHasInstance(ctx *controllerContext, groupLocation, groupName string, instanceID uint64) (bool, error) {
 	recordMetric := csrmetrics.OutboundRPCStartRecorder("compute.InstanceGroupManagersService.ListManagedInstances")
-	instances, err := compute.NewInstanceGroupManagersService(ctx.gcpCfg.Compute).ListManagedInstances(ctx.gcpCfg.ProjectID, groupLocation, groupName).Do()
-	if err != nil {
-		recordMetric(csrmetrics.OutboundRPCStatusError)
-		return false, err
+	filter := func(response *compute.InstanceGroupManagersListManagedInstancesResponse) error {
+		for _, instance := range response.ManagedInstances{
+			if instance.Id == instanceID {
+				return &foundError{}
```

Comment:
We are listing all the instances and going through them page by page when we find the instance we are looking for we error out to early exit. 
Yes this is quirk of pagination where it can only be stopped by returning an error. So I created a fake error to differentiate this case. 

The list api does not support filtering so this hacky pagination early exit way is what we are using for now.

Reply:
Ah okay that makes sense.

I would add a comment to explain this hack.

Reply:
Done. Added the comment.